### ISSUE TITLE
Unexpected loss of iptables rule, adding again

### ISSUE THREAD

**Issue by monobilisim** [pritunl/pritunl#323]
On our  Pritunl Server we connect some of client's VPN networks via OpenVPN Client. We use Debian 8 and we put client conf files to /etc/openvpn and we generate services like "systemctl enable openvpn@vpn_dest_name.service".

Let me give an example: openvpn@vpn_dest_name.service has 10.34.0.0/16 network behind it. And we submit a route to our Pritunl VPN Server for 10.34.0.0/16. We have about 5 Pritunl VPN Servers and our Pritunl Server. We  also add routes for internal  Pritunl VPN Servers. Here is a screenshot of our one of Pritunl VPN Server which contains routes for both other Pritunl  Servers & /etc/openvpn OpenVPN Clients'.

![](http://i.imgur.com/Bq1KKVF.png)

After upgrading to 1.21.942.24-0debian1~jessie routes do not work correctly. Before 1.21 upgrade, after a clean reboot I can reach both Pritunl VPN Server network (198.18.0.0/16) which is on the same server and other OpenVPN Client networks which config's are stored in /etc/openvpn. But after 1.21, after clean reboot and I need to stop & start Pritunl VPN Server to make all routes work.

How do we need to handle this issue? Thank you so much. 

```
[thawing-plateau-4721][2016-04-15 17:00:18,381][ERROR] Unexpected loss of iptables rule, adding again...
  rule = ['OUTPUT', '-o', 'tun5', '-s', '198.18.0.1', '-j', 'ACCEPT']
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 783, in __bootstrap
    self.__bootstrap_inner()
  File "/usr/lib/python2.7/threading.py", line 810, in __bootstrap_inner
    self.run()
  File "/usr/lib/python2.7/threading.py", line 763, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/usr/lib/pritunl/local/lib/python2.7/site-packages/pritunl/server/instance.py", line 639, in _iptables_thread
    self.iptables.upsert_rules(log=True)
  File "/usr/lib/pritunl/local/lib/python2.7/site-packages/pritunl/iptables.py", line 508, in upsert_rules
    rule=rule,
  File "/usr/lib/pritunl/local/lib/python2.7/site-packages/pritunl/logger/__init__.py", line 55, in error
    kwargs['traceback'] = traceback.format_stack()
```

**Comment by zachhuff386** [comment]
Disable _Restrict Routes_ in server settings. Using user network links will allow setting up iroutes and will avoid this issue.

**Comment by monobilisim** [comment]
I disabled "Restrict Routes" for all Pritunl VPN Servers. But I still receive this errors:

```
[thawing-plateau-4721][2016-04-16 12:33:29,369][ERROR] Popen returned error exit code
  cmd         = ['iptables', '-I', 'INPUT', '-i', 'tun5', '-d', u'192.168.132.0/23', '-j', 'ACCEPT', '-m', 'comment', '--comment', 'pritunl_56c98b31dcff71767cdfe948']
  return_code = 4
Process stderr:
  Another app is currently holding the xtables lock. Perhaps you want to use the -w option?
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 783, in __bootstrap
    self.__bootstrap_inner()
  File "/usr/lib/python2.7/threading.py", line 810, in __bootstrap_inner
    self.run()
  File "/usr/lib/python2.7/threading.py", line 763, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/usr/lib/pritunl/local/lib/python2.7/site-packages/pritunl/server/instance.py", line 639, in _iptables_thread
    self.iptables.upsert_rules(log=True)
  File "/usr/lib/pritunl/local/lib/python2.7/site-packages/pritunl/iptables.py", line 510, in upsert_rules
    self._insert_iptables_rule(rule)
  File "/usr/lib/pritunl/local/lib/python2.7/site-packages/pritunl/iptables.py", line 465, in _insert_iptables_rule
    ['ip6tables' if ipv6 else 'iptables', '-I'] + rule)
  File "/usr/lib/pritunl/local/lib/python2.7/site-packages/pritunl/utils/misc.py", line 146, in check_output_logged
    stderr=stderrdata,
  File "/usr/lib/pritunl/local/lib/python2.7/site-packages/pritunl/logger/__init__.py", line 55, in error
    kwargs['traceback'] = traceback.format_stack()
[thawing-plateau-4721][2016-04-16 12:33:29,370][ERROR] Failed to insert iptables rule, retrying...
  rule = ['INPUT', '-i', 'tun5', '-d', u'192.168.132.0/23', '-j', 'ACCEPT', '-m', 'comment', '--comment', 'pritunl_56c98b31dcff71767cdfe948']
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 783, in __bootstrap
    self.__bootstrap_inner()
  File "/usr/lib/python2.7/threading.py", line 810, in __bootstrap_inner
    self.run()
  File "/usr/lib/python2.7/threading.py", line 763, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/usr/lib/pritunl/local/lib/python2.7/site-packages/pritunl/server/instance.py", line 639, in _iptables_thread
    self.iptables.upsert_rules(log=True)
  File "/usr/lib/pritunl/local/lib/python2.7/site-packages/pritunl/iptables.py", line 510, in upsert_rules
    self._insert_iptables_rule(rule)
  File "/usr/lib/pritunl/local/lib/python2.7/site-packages/pritunl/iptables.py", line 473, in _insert_iptables_rule
    rule=rule,
  File "/usr/lib/pritunl/local/lib/python2.7/site-packages/pritunl/logger/__init__.py", line 55, in error
    kwargs['traceback'] = traceback.format_stack()
``[thawing-plateau-4721][2016-04-16 12:33:29,369][ERROR] Popen returned error exit code
  cmd         = ['iptables', '-I', 'INPUT', '-i', 'tun5', '-d', u'192.168.132.0/23', '-j', 'ACCEPT', '-m', 'comment', '--comment', 'pritunl_56c98b31dcff71767cdfe948']
  return_code = 4
Process stderr:
  Another app is currently holding the xtables lock. Perhaps you want to use the -w option?
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 783, in __bootstrap
    self.__bootstrap_inner()
  File "/usr/lib/python2.7/threading.py", line 810, in __bootstrap_inner
    self.run()
  File "/usr/lib/python2.7/threading.py", line 763, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/usr/lib/pritunl/local/lib/python2.7/site-packages/pritunl/server/instance.py", line 639, in _iptables_thread
    self.iptables.upsert_rules(log=True)
  File "/usr/lib/pritunl/local/lib/python2.7/site-packages/pritunl/iptables.py", line 510, in upsert_rules
    self._insert_iptables_rule(rule)
  File "/usr/lib/pritunl/local/lib/python2.7/site-packages/pritunl/iptables.py", line 465, in _insert_iptables_rule
    ['ip6tables' if ipv6 else 'iptables', '-I'] + rule)
  File "/usr/lib/pritunl/local/lib/python2.7/site-packages/pritunl/utils/misc.py", line 146, in check_output_logged
    stderr=stderrdata,
  File "/usr/lib/pritunl/local/lib/python2.7/site-packages/pritunl/logger/__init__.py", line 55, in error
    kwargs['traceback'] = traceback.format_stack()
[thawing-plateau-4721][2016-04-16 12:33:29,370][ERROR] Failed to insert iptables rule, retrying...
  rule = ['INPUT', '-i', 'tun5', '-d', u'192.168.132.0/23', '-j', 'ACCEPT', '-m', 'comment', '--comment', 'pritunl_56c98b31dcff71767cdfe948']
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 783, in __bootstrap
    self.__bootstrap_inner()
  File "/usr/lib/python2.7/threading.py", line 810, in __bootstrap_inner
    self.run()
  File "/usr/lib/python2.7/threading.py", line 763, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/usr/lib/pritunl/local/lib/python2.7/site-packages/pritunl/server/instance.py", line 639, in _iptables_thread
    self.iptables.upsert_rules(log=True)
  File "/usr/lib/pritunl/local/lib/python2.7/site-packages/pritunl/iptables.py", line 510, in upsert_rules
    self._insert_iptables_rule(rule)
  File "/usr/lib/pritunl/local/lib/python2.7/site-packages/pritunl/iptables.py", line 473, in _insert_iptables_rule
    rule=rule,
  File "/usr/lib/pritunl/local/lib/python2.7/site-packages/pritunl/logger/__init__.py", line 55, in error
    kwargs['traceback'] = traceback.format_stack()
```

I just rebooted the server. And reconnect to Pritunl VPN Server.  I can ping 192.168.132.3 (which is behind openvpn@doganhukuk.service which conf file is located at /etc/openvpn) but I cannot ping 198.18.0.3 which is behind one of Pritunl VPN Servers. I can ping 198.18.0.1.

After stopping and starting and reconnecting Pritunl VPN Server which I was connected, I can both ping  198.18.0.3, 192.168.132.3 and 198.18.0.1.

**Comment by zachhuff386** [comment]
There was an issue with disabling the route restrictions, it is fixed in v1.21.947.73
